### PR TITLE
Add Dart Code Metrics (DCM) linter; change code to comply with new rules

### PR DESCRIPTION
**Pull request by TTA777**
Added the DCM linter in addition to our current one, with fixes for the rules that we were currently breaking.

The preset enabling all rules was chosen, though several rules have been disabled from the preset, some of which we will likely want to add in the future.

### CODE REVIEW

File: lib/service_locator.dart [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// Simplifies usage of service locator greatly.
+//ignore_for_file: no-equal-arguments
```
Comment (marfavi) [inline lib/service_locator.dart:2]: How does ignoring this rule help simplify the usage of sl? I don't see any linter errors by removing this line

File: analysis_options.yaml [hunk 2]
```diff
@@ -17,10 +17,47 @@ analyzer:
...
+    - avoid-returning-widgets: false
+    - avoid-late-keyword: false
+    - prefer-single-widget-per-file: false
+    - avoid-passing-async-when-sync-expected: false
+    - avoid-redundant-async: false
+    - avoid-non-ascii-symbols: false
+    - prefer-match-file-name: false
+    - avoid-shrink-wrap-in-lists: false
+    - no-equal-arguments: false
+    # - avoid-ignoring-return-values
```
Comment (marfavi) [inline analysis_options.yaml:61]: Add a line explaining why these are commented out

**Review by marfavi — CHANGES_REQUESTED** [review]
Small comments. Also, `make analyze` gives a warning:

```
omid@Omids-MacBook-Pro coffeecard_app % make analyze
flutter analyze && \
        flutter pub run dart_code_metrics:metrics analyze lib
Analyzing coffeecard_app...                                             
No issues found! (ran in 5.6s)
✔ Analysis is completed. Preparing the results: 11.4s

lib/cubits/statistics/statistics_cubit.dart:
WARNING LeaderboardCubit.fetch
        maximum nesting level: 5

```

**Review by marfavi — APPROVED** [review]

File: lib/widgets/pages/settings/your_profile_page.dart [hunk 3]
```diff
@@ -67,7 +68,7 @@ class _EditProfile extends StatelessWidget {
         ),
         const Gap(8),
         Text(
-          '${user.occupation.fullName} (${user.occupation.shortName})',
+          '${user.occupation.fullName} ($occupationShotName)',
```
Comment (fremartini) [inline lib/widgets/pages/settings/your_profile_page.dart:71]: occupationShortName

**Review by fremartini — APPROVED** [review]

**Review by fremartini — APPROVED** [review]

**Pull request merged**